search-sentiments-map/routes/search.js

```diff
@@ -0,0 +1,278 @@
+  const [searchResults] = await datastore.runQuery(query);
+
+  // Note: Can't use forEach with await.
+  // Loop through sorted data beginnning with oldest results, delete if older
+  // than a week. Stop when reach results from within a week.
+  for (let i = 0; i < searchResults.length; i++) {
+    if (Date.now() - searchResults[i].timestamp > 7 * 24 * 60 * 60000) {
+      const searchResultKey = searchResults[i][datastore.KEY];
+      await datastore.delete(searchResultKey);
+      console.log(`Custom Search Result ${searchResultKey.id} deleted.`)
+    } else {
+      break;
+    }
+  }
+}
+
+/** 
+ * Creates Datastore item for given topic and country search results. 
+ * @param {string} topic The seach topic the search results are for.
+ * @param {Object} countriesData Object holding all searchResults for all
+ *      countries.
+ */
+async function addTopicToDatastore(topic, countriesData) {
+  const customSearchTopicKey = datastore.key('CustomSearchTopic');
+  // Get the current timestamp in milliseconds.
+  let timestamp = Date.now();
+  const entity = {
+    key: customSearchTopicKey,
+    data: {
+      topic: topic,
+      countries: countriesData,
+      timestamp: Date.now(),
+    },
+  };
+  try {
+    await datastore.save(entity);
+    console.log(`Custom Search Result ${customSearchTopicKey.id} created successfully.`);
+  } catch (err) {
+    console.error('ERROR:', err);
+  }
+}
+
+/**
+ * Adds search result object to countryData array.
+ * @param {Object} searchResult Object with information for one search result.
+ * @param {Object} countryData Object holding all searchResults for a country.
+ * @return {Promise} A Promise wrapped around the sentiment score of the result.
+*/
+function addSearchResultToCountryData(searchResult, countryData) {
+  return getSentiment(searchResult)
+    .then(response => response.json())
+    .then((result) => {
+      searchResultData = {
+        title: searchResult.title,
+        snippet: searchResult.snippet,
+        htmlTitle: searchResult.htmlTitle,
+        link: searchResult.link,
+        score: result.score,
+      };
+      countryData.push(searchResultData);
+      return result.score;
+    });
+
+}
+
+// TODO(ntarn): Look into changing the above method so that it does not need to return 
+// twice for a Promise wrapped around a object.
+// async function addSearchResultToCountryData(searchResult, countryData) {
+//   let sentimentScore;
+//   await getSentiment(searchResult)
+//     .then(response => response.json())
+//     .then((result) => { 
+//       console.log('ntarn debug: frontend' + result.score);
+//       searchResultData = {
+//         title: searchResult.title,
+//         snippet: searchResult.snippet,
+//         htmlTitle: searchResult.htmlTitle,
+//         link: searchResult.link,
+//         score: result.score,
+//       };
+//       countryData.push(searchResultData);
+//       sentimentScore = result.score;
+//     });
+//   return sentimentScore;
+// }
```

Comment:
Is this still needed?

Reply:
Working on making better code so there's not two return statements if posssible